### DISCUSSION TITLE
case insensitive contains -> icontains

### DISCUSSION THREAD

**Discussion by matiasrebori** [sqlalchemy/sqlalchemy#8287] [Ideas]
### Suggestion
hello, i was struggling to find a way to make a query that returns every row with a specific column that has a substring of a specific string. i finded `like` and `ilike` operators but this makes an exact match if you dont add the  `LIKE wildcards` so it wasnt working straight away for my use ase, then i looked in the docs and foundthat i could use the `like method` with the `LIKE wildcards` to make an `isensitive contains -> icontains`.
since the `like` operator has its insensitive version, I think it is a good idea that the `contains` operator also has its insensitive version. contains operator [contains docs](https://docs.sqlalchemy.org/en/14/core/sqlelement.html#sqlalchemy.sql.expression.ColumnOperators.contains) is an implementation of `LIKE` operator
### Reason
this will make the use of SQLAlchemy a bit more easy, also it will improve the consistency in the methods, it will be  alike as `django ORM filter icontains` making the transition from django to SQLAlchemy more easy

**Comment by zzzeek** [comment]
this is covered by #3482. it's been stalled for many years due to lack of contributor interest.

**Comment by matiasrebori** [comment]
@zzzeek i could try to submit a PR if its ok, it seems fairly easy compared with SQLAlchemy overall complexity.

**Reply by CaselIT** [reply]
Sure, go ahead. Feel free to ask suggestions or questions on the implementation in the issue linked above
